app/src/main/java/eu/theinvaded/mastondroid/viewmodel/LoginViewModelContract.java

```diff
@@ -20,7 +20,7 @@
         String getClientSecret();
         String getClientId();
         void setUser(MastodonAccount account);
-        void showLoginError();
+        void showLoginError(Throwable loginError);
```

Comment:
Yeah, okay, I will. It's one of the principles of MVVM: do not pass actual models to the View (or anything that View shouldn't care about). Better stick to the primitives when working with the View layer so view cannot screw up anything and contract stays flexible.